### PR TITLE
Fixes #235 - Add docker push to travis.yml

### LINKED ISSUE #235: Release Docker images upon tag pushed
* Upon successful builds, release a Docker image with the corresponding tag.

This should cover openfaas/faas, openfaas/faas-netes and openfaas/nats-queue-worker ideally via Travis or Docker Hub.

### PR DESCRIPTION
**Pull request by BurtonR**
Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

## Description
Added push to Docker Hub on git tag with image tagging

Must set the Travis-CI environment variables on the project:
DOCKER_USERNAME and DOCKER_PASSWORD
in order to log in to docker for the push. Make sure the setting 'Display value in build log' is disabled
This will run when a git tag is pushed into the repository. It will push one image with the git tag value

## Motivation and Context
Hacktoberfest!

## How Has This Been Tested?
`git tag -a 1.0 -m "Tagging the commit"`
`git push origin 1.0`
then watch Travis do the work!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,3 +14,11 @@ script:
     - sh build.sh
     # Invoke ci script too
     - sh contrib/ci.sh
+after_success:
+    - if [[ "$TRAVIS_TAG" != null ]]; then
```
Comment (alexellis) [inline .travis.yml]: During testing this did not work for me. This appears to work:

```
if [ ! -s "$TRAVIS_TAG" ] ; then
    echo "This build will be published under the tag: ${TRAVIS_TAG}"
fi
```

File: .travis.yml [hunk 2]
```diff
@@ -14,3 +14,11 @@ script:
     - sh build.sh
     # Invoke ci script too
     - sh contrib/ci.sh
+after_success:
+    - if [[ "$TRAVIS_TAG" != null ]]; then 
+        docker tag functions/gateway:latest-dev $DOCKER_USERNAME/gateway:latest;
```
Comment (alexellis) [inline .travis.yml]: I think we said we wouldn't push `:latest` in the previous comments.

File: .travis.yml (hunk 1 above)
Comment (BurtonR) [reply .travis.yml]: Strange. I tested it with my Travis account and it worked fine, but I updated it to match what you show

File: .travis.yml (hunk 2 above)
Comment (BurtonR) [reply .travis.yml]: We did discuss it, but wasn't sure if we came to a final resolution. I suppose, for now at least, we won't push a `:latest` tag in Travis. We may come back and add that later

**Comment by BurtonR** [comment]
@alexellis Would you mind having another look at this?

**Pull request merged**